gdnative-core/src/core_types/variant.rs

```diff
@@ -1774,6 +1803,38 @@ godot_test!(
         assert_eq!(Some(&f64::to_variant(&54.0)), vec_maybe[2].as_ref().err());
     }
 
+    test_variant_hash_map {
+        let original_hash_map = HashMap::from([
+            ("Foo".to_string(), 4u32),
+            ("Bar".to_string(), 2u32)
+        ]);
+        let variant = original_hash_map.to_variant();
+        let check_hash_map = variant.try_to::<HashMap<String, u32>>().expect("should be hash map");
+        assert_eq!(original_hash_map, check_hash_map);
+        // Check conversion of heterogeneous dictionary key types
+        let non_homogenous_key_dictonary = Dictionary::new();
+        non_homogenous_key_dictonary.insert("Foo".to_string(), 4u32);
+        non_homogenous_key_dictonary.insert(7, 2u32);
+        assert_eq!(
+            Err(FromVariantError::InvalidVariantType {
+                variant_type : VariantType::I64,
+                expected: VariantType::GodotString
+            }),
+            non_homogenous_key_dictonary.owned_to_variant().try_to::<HashMap<String, u32>>()
+        );
+        // Check conversion of heterogeneous dictionary value types
+        let non_homogenous_value_dictonary = Dictionary::new();
+        non_homogenous_value_dictonary.insert("Foo".to_string(), 4u32);
+        non_homogenous_value_dictonary.insert("Bar".to_string(), "Unexpected".to_string());
+        assert_eq!(
+            Err(FromVariantError::InvalidVariantType {
+                variant_type : VariantType::GodotString,
+                expected: VariantType::I64
+            }),
+            non_homogenous_value_dictonary.owned_to_variant().try_to::<HashMap<String, u32>>()
```

Comment:
Maybe it's worth if we go through some of these (outside the scope of this PR), but you can gladly change this specific occurrence here.

Rust itself doesn't mention "expected" and "actual" in the [`assert_eq!` docs](https://doc.rust-lang.org/std/macro.assert_eq.html), but rather "left" and "right", seemingly indicating freedom of choice. 

_**However**_, in its own docs, Rust often uses `assert_eq!(actual, expected)`, e.g. for [`Vec`](https://doc.rust-lang.org/std/vec/struct.Vec.html#examples-16). I find that more intuitive, too, because it corresponds to the human way of thinking "_this_ should have _that_ value", rather than "_this_ value should be in _that_".